fairscale/optim/feed_forward2.py

```diff
@@ -0,0 +1,204 @@
+import logging
+from typing import Tuple
+
+import numpy as np
+from sklearn.datasets import make_blobs
+import torch
+import torch.nn as nn
+
+
+class Feedforward(torch.nn.Module):
+    def __init__(self, input_size: int, hidden_size: int):
+        torch.manual_seed(7)
+        super(Feedforward, self).__init__()
+        self.input_size = input_size
+        self.hidden_size = hidden_size
+        self.fc1 = nn.Linear(self.input_size, self.hidden_size)
+        self.relu = nn.ReLU()
+        self.fc2 = nn.Linear(self.hidden_size, 1)
+        self.sigmoid = nn.Sigmoid()
+
+    def forward(self, x: torch.Tensor) -> torch.Tensor:  # type: ignore
+        out = self.fc1(x)
+        out = self.relu(out)
+        out = self.fc2(out)
+        out = self.sigmoid(out)
+        return out
+
+
+class LayerwiseGradientScaler:
+    def __init__(self, modules: nn.Module):
+        self.per_layer_scale = [2 ** (1 / 1000) for _ in modules]
+        # print('length of modules %s' % len(modules))
+
+    def scale_up(self, m: nn.Module, inputs: torch.Tensor, outputs: torch.Tensor) -> None:
+        if inputs[0] is not None:
+            torch.mul(inputs[0], (2 ** 13))
+
+    def scale_down(self, m: nn.Module, inputs: torch.Tensor, outputs: torch.Tensor) -> None:
+        if outputs[0] is not None:
+            torch.mul(outputs[0], 1 / (2 ** 13))
```

Comment:
Right now it does not associate the per_layer_scale to the module itself, perhaps consider storing the modules (or some reference to them) in a list during init, and then using this variable to pull the corresponding layer_scale